README.md

```diff
@@ -1 +1,299 @@
+
+### Test Orders ###
+
+For testing purposes, Amazon DRS allows making [test orders](https://developer.amazon.com/docs/dash/test-device-purchases.html). Test orders are those that made by a DRS device authenticated as a test device. So it is determined at the step of authentication whether the device is for testing or not.
+Due to this the [login()](#loginrocky-devicemodel-deviceserial-onauthenticated-testdevice) method has a parameter *testDevice*. But if you set a `Refresh Token` manually with the [setRefreshToken()](#setrefreshtokenrefreshtoken) method, only you know whether this token was obtained for testing or not and such *testDevice* parameter is not required here.
+
+Only test orders can be canceled with the [cancelTestOrder()](#canceltestorderslotid-oncanceled) method.
+
+### Callbacks ###
+
+All requests that are made to the Amazon platform occur asynchronously. Every method that sends a request has an optional parameter which takes a callback function that will be executed when the operation is completed, whether successfully or not. The callback’s parameters are listed in the corresponding method description, but every callback has at least one parameter, *error*. If *error* is `0`, the operation has been executed successfully. Otherwise, *error* is a code of an error.
+
+## AmazonDRS Class ##
+
+### Constructor: AmazonDRS(*clientId, clientSecret*) ###
+
+This method returns a new AmazonDRS instance.
+
+| Parameter | Data Type | Required? | Description |
+| --- | --- | --- | --- |
+| *clientId* | String | Yes | `Client ID` of your LWA Security Profile. For information, please see [here](https://developer.amazon.com/docs/login-with-amazon/glossary.html#client_identifier). |
+| *clientSecret* | String | Yes | `Client Secret` of your LWA Security Profile. For information, please see [here](https://developer.amazon.com/docs/login-with-amazon/glossary.html#client_secret). |
+
+### login(*rocky, deviceModel, deviceSerial[, onAuthenticated[, testDevice]]*) ###
+
+This method allows to authenticate the agent on the Amazon and get required security tokens. The method automatically sets the obtained tokens to be used for DRS API calls, so you do not need to call the [setRefreshToken()](#setrefreshtokenrefreshtoken) method. For more information, please read about [authentication](#authentication). 
+
+You may call this method again only after the previous call has finished. The call is considered as finished only when the authentication flow described in the [authentication](#authentication) section is completed or an error occured.
+
+Either this method or [setRefreshToken()](#setrefreshtokenrefreshtoken) should be called and authentication should be done before making any DRS-related requests.
+
+This method uses the [Rocky library](https://github.com/electricimp/Rocky), so it requires an instance of Rocky to be passed in.
```

Comment:
Add a bit more here about the route that Rocky uses and what users will need to do to avoid conflicts. 
Might be worth letting users define a custom route if their application uses Rocky in the application flow?

Reply:
Yes, it all makes sense!
Thank you!